### DISCUSSION TITLE
Why doesn't markdoc support for loops?

### DISCUSSION THREAD

**Discussion by louiss0** [markdoc/markdoc#350] [Q&A] [answered]
I looked into the docs and found out that loops are not supported in markdoc. I'd like to create a tag for this but unfortunately, I can't use a function as children. Why are loops simply not supported? Markdoc is a templating language.

**Comment by rpaul-stripe** [comment, marked as answer]
This is an intentional choice that was informed by the challenges that we faced using conventional templating systems in the past. Markdoc was designed explicitly to decouple code from content, encouraging users to move procedural content generation out of the document itself and encapsulate it in code—either in a custom tag definition or a React component. Putting this logic in code instead of the document means that it is more testable, maintainable, and easier to localize.

In cases where we want to generate a table from data in the document, for example, we typically make a custom tag that encapsulates the the logic for generating the table and we pass in the data as an array attribute value. We have a simple example of this in the Markdoc documentation: https://markdoc.dev/docs/examples#loops
